specs/TryUpdateRuntime.md

```diff
@@ -0,0 +1,225 @@
+  /// normally, regardless of whether we could update the Edge WebView2
+  /// Runtime. If an unexpected error interrupts the update operation, error
+  /// code of that unexpected error would be set as `errorCode`.
+  /// When update operation can be performed normally, but update resulted in
+  /// failure, like download failed, the error code would be presented as
+  /// `UpdateError` property of ICoreWebView2ExperimentalUpdateRuntimeResult.
+  HRESULT Invoke([in] HRESULT errorCode,
+                 [in] ICoreWebView2ExperimentalUpdateRuntimeResult * result);
+}
+
+/// This interface is an extension of the ICoreWebView2Environment. An object
+/// implementing the ICoreWebView2ExperimentalEnvironment3 interface will also
+/// implement ICoreWebView2Environment.
+[uuid(9A2BE885-7F0B-4B26-B6DD-C969BAA00BF1), object, pointer_default(unique)]
+interface ICoreWebView2ExperimentalEnvironment3 : IUnknown {
+  /// Try to update installed Microsoft Edge WebView2 Runtime.
+  /// This will potentially result in a new version of the Edge Webview2
+  /// Runtime being installed and `NewBrowserVersionAvailable` event fired.
+  /// There is no guarantee on the order of that event being fired and
+  /// TryUpdateRuntimeis completed handler being invoked. Besides the
+  /// `NewBrowserVersionAvailable` event, there will be no impact to any
+  /// currently running WebViews when update is installed.
+  /// The latest version can always be queried using
+  /// `GetAvailableCoreWebView2BrowserVersionString` API.
+  /// This is only supported for installed Edge WebView2 Runtime. When running
+  /// fixed version Edge WebView2 Runtime or non stable channel Edge browser,
+  /// this API will return `HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED)`.
+  /// There could only be one active TryUpdateRuntime operation, calling the API
+  /// before completed handler for previous call is invoked will fail with
+  /// `HRESULT_FROM_WIN32(ERROR_BUSY)`.
+  /// Calling the API repeatedly in a short period of time, will also fail with
+  /// `HRESULT_FROM_WIN32(ERROR_BUSY)`.
+  /// The TryUpdateRuntime operation is associated with the CoreWebView2Environment
+  /// object, on going TryUpdateRuntime operation will be aborted when the
+  /// associated CoreWebView2Environment along with the CoreWebView2 objects that
+  /// are created by the CoreWebView2Environment object are all released.
+  ///
+  /// \snippet AppWindow.cpp UpdateRuntime
+  HRESULT TryUpdateRuntime(
+      [in] ICoreWebView2ExperimentalTryUpdateRuntimeCompletedHandler *
+      handler);
+}
+
+```
+## .NET WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    public enum CoreWebView2RuntimeUpdateStatus
+    {
+        NoUpdate = 0,
+        Updated = 1,
+        BlockedByPolicy = 2,
+        Failed = 3,
+    }
+
+    public partial class CoreWebView2UpdateRuntimeResult
+    {
+        public CoreWebView2RuntimeUpdateStatus UpdateRuntimeStatus
```

Comment:
the type is RuntimeUpdateStatus and the property name is UpdateRuntimeStatus. Probably want to be consistent for both. Or name the property just 'Status'

Reply:
Oh also I guess the type name is inconsistent with the others:

TryUpdateRuntimeAsync
UpdateRuntimeResult
RuntimeUpdateStatus

Can that be UpdateRuntimeStatus?

Reply:
Will change to UpdateRuntimeStatus and name the property as Status.